lib/chainer/functions/array/broadcast_to.rb

```diff
@@ -0,0 +1,43 @@
+module Chainer
+  module Functions
+    module Array
+      # Function that broadcasts an array to a new shape."""
+      class BroadcastTo < Function
+        def initialize(shape)
+            @shape = shape
+        end
+
+        def self.broadcast_to(xs, shape)
+          self.new(shape).(xs)
+        end
+
+        def forward(xs)
```

Comment:
When using the `forward` function it is necessary to initialize with `retain_inputs`
ref. https://github.com/red-data-tools/red-chainer/blob/master/lib/chainer/functions/activation/relu.rb#L30

Reply:
Why is there such an implicit rule?
It should be in caller of `foward` if always need it.

Reply:
Please See below. 

https://github.com/chainer/chainer/blob/v2.0.2/chainer/function.py#L127
or
https://docs.chainer.org/en/v2.0.2/reference/core/generated/chainer.Function.html#chainer.Function

And, I think that it is better to reflect this content in the comment of lib/chainer/function.rb.

Reply:
Thank you! I see.
I suggest it is not necessary to call `retain_input` when inputs is not used in backward. We should not inherit bad interface from chainer.

Reply:
With the change to Chainer 3.0 ("the new-style differentiable function") of Red-Chainer base code in the future, I think that initialization processing of retain_inputs becomes unnecessary.

Please read the following.
https://files.gitter.im/red-data-tools/ja/IN8R/IMG_5654.JPG
https://github.com/chainer/chainer/pull/2970
https://github.com/chainer/chainer/pull/3420

Reply:
> With the change to Chainer 3.0 ("the new-style differentiable function") of Red-Chainer base > code in the future, I think that initialization processing of retain_inputs becomes unnecessary.

yes.
The base code of red-chainer is currently proceeding to correspond to Chainer 3.0.
Now We are porting Chainer's code as it is, but I'd like to gradually improve it.